salt/docker/init.sls

```diff
@@ -16,6 +16,7 @@ include:
         Environment="NO_PROXY={{ salt['pillar.get']('proxy:no_proxy', '') }}"
   cmd.run:
     - name: systemctl daemon-reload
+    - stateful: True
```

Comment:
Just checked it: using `/etc/systemd/system/docker.service.d/proxy.conf` directly instead of `file: /etc/systemd/system/docker.service.d/proxy.conf` ensures that this runs also if the file changes (even with the `cmd.run` on that state), and the order is the correct one on all my tests.

No changes -> nothing is done
Changes -> write changes, daemon-reload, restart docker